src/main/java/ga/rugal/food/core/service/impl/MenuServiceImpl.java

```diff
@@ -50,5 +53,17 @@ public Menu deleteById(Integer id)
     {
         return menuDao.deleteById(id);
     }
+    
+    @Override
+    public List<Menu> getByRestaurant(Restaurant r) {
+        return menuDao.getByRestaurant(r);
+    }
+    
+    @Override
+    public Menu getRandomMenuByRetaurant(List<Menu> list) {
```

Comment:
This method is weird! Please refactor it.  
1. Please clarify its functionality.  
2. Please think clearly about design pattern and code discipline.